std/experimental/logger/core.d

```diff
@@ -188,39 +188,27 @@ Params:
   ll = The `LogLevel` used by this log call.
   condition = The condition must be `true` for the data to be logged.
   args = The data that should be logged.
+  line = auto-filled with __LINE__ macro
+  file = auto-filled with __FILE__ macro
+  funcName = auto-filled with __FUNCTION__ macro
+  prettyFuncName = auto-filled with __PRETTY_FUNCTION__ macro
 
 Example:
 --------------------
 log(LogLevel.warning, true, "Hello World", 3.1415);
 --------------------
 */
-void log(int line = __LINE__, string file = __FILE__,
-    string funcName = __FUNCTION__, string prettyFuncName = __PRETTY_FUNCTION__,
-    string moduleName = __MODULE__, A...)(const LogLevel ll,
-    lazy bool condition, lazy A args)
-if (args.length != 1)
-{
-    static if (isLoggingActive)
-    {
-        if (ll >= moduleLogLevel!moduleName)
-        {
-               stdThreadLocalLog.log!(line, file, funcName, prettyFuncName, moduleName)
-                (ll, condition, args);
-        }
-    }
-}
-
-/// Ditto
-void log(T, string moduleName = __MODULE__)(const LogLevel ll,
-    lazy bool condition, lazy T arg, int line = __LINE__, string file = __FILE__,
-    string funcName = __FUNCTION__, string prettyFuncName = __PRETTY_FUNCTION__)
+void log(string moduleName = __MODULE__, A...)(const LogLevel ll,
+    lazy bool condition, lazy A args, int line = __LINE__,
```

Comment:
I'm surprised this is legal in D, as it's impossible to actually set `line` manually.

Reply:
It's only been allowed since 2.079 and it seems to have been added to the language [specifically for this function](https://dlang.org/changelog/2.079.0.html#default_after_variadic).